test/exercises/wavelet/wavelet.chpl

```diff
@@ -0,0 +1,66 @@
+use ecgIO;
+import Math.log2;
+
+config const nLevels = 3;
+config param plotResults = false;
+
+proc main() {
+    // read sample data
+    //  (data pulled from: http://paulbourke.net/dataformats/holter/)
+    const ecgSamples = try! readEcgData("ecg_samples.txt"),
+          lenP2 = 2**(floor(log2(ecgSamples.size)):int);
+
+    // compute 1D DWT
+    const ecg_dwt = haarWavelet1D(ecgSamples[0..<lenP2], nLevels);
+
+    // generate a plot of the signal and DWT
+    if plotResults then plotDwtData(ecgSamples[0..<lenP2], ecg_dwt, nLevels);
+}
+
+// compute an n-level wavelet transform of 'x'
+proc haarWavelet1D(x: [?d] ?t, n: int) : [d] t
+    where d.rank == 1 && isNumericType(t)
+{
+    assert(2**log2(d.size) == d.size, "array size must be a power of 2");
+
+    var output : [d] t;
+    hwRec(x, output, d.size, d.size / 2**n);
+    return output;
+}
+
+// recursive helper for 'haarWavelet1D'
+proc hwRec(const signal, ref output, fmax: int, fstop: int) {
+    if fmax == fstop {
+        // store the final layer of high-pass coefficients
+        output[{0..<fmax}] = signal;
```

Comment:
I'd drop the curly brackets from this slice as they'll add a minor amount of execution-time overhead, and also likely raise questions about what they mean relative to `output[0..<fmax]`.  (They do mean different things for distributed arrays, but for local ones like this one it's a wash—and even in the distributed case, I'd say the non-curly-bracket case is the better one to have people use by default).